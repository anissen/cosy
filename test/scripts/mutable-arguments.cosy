
{
    struct S {
        mut x = 0
        let y = 1
    }
    // print S {}

    // struct S2 {
    //     mut y = 0
    // }
    // print S2 {}

    // let s = S { x = 1 }
    mut s = S { x = 2, y = 3 }
    mut s2
    s2 = s
    print s

    fn mutable_param(mut p S) {
        p.x = 4
        // p.y = 5
        print p
    }
    // mutable_param(s)
    mutable_param(mut s2)
    print s
}

{
    struct Point {   // Struct
        mut x
        let y // TODO: What to do about struct members of unknown type?
        let text = 'default value'
    }
    // instantiate struct
    mut point = Point { x = 3, y = 4 }
    // mut point = Point { x = '3', y = 4 }
    print point

    fn mutf(mut p Point) {
        p.x = 5
    }
    print point
    mutf(point) // TODO: Fail! Should be mut
    print point
}
