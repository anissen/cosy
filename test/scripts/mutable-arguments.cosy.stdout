S { x = 2, y = 3 }
S { x = 4, y = 3 }
S { x = 4, y = 3 }
Point { text = "default value", x = 3, y = 4 }
Point { text = "default value", x = 3, y = 4 }
Point { text = "default value", x = 5, y = 4 }
